channelserver/src/main.rs

```diff
@@ -112,35 +114,61 @@ fn channel_route(req: &HttpRequest<session::WsChannelSessionState>) -> Result<Ht
     )
 }
 
-/// Inject the FoxSec headers into all HTTP responses
-fn add_headers<'a>(resp: &'a mut HttpResponseBuilder) -> &'a mut HttpResponseBuilder {
-    resp.header(
-        "Content-Security-Policy",
-        "default-src 'none'; img-src 'self'; script-src 'self'; style-src 'self'; report-uri /__cspreport__",
-    )
-    .header("X-Content-Type-Options", "nosniff")
-    .header("X-Frame-Options", "deny")
-    .header("X-XSS-Protection", "1; mode=block")
-    .header("Strict-Transport-Security", "max-age=63072000")
+struct Headers;
+
+impl<S> Middleware<S> for Headers {
+    /*
+    // Currently extracting the meta info only when building a channel, rather than
+    // on every request. If that's preferred, use this function:
+    fn start(&self, req: &HttpRequest<S>) -> Result<Started, Error> {
+        Ok(Started::Done)
+    }
+    */
+
+    /// Set the OpSec headers on all responses.
+    fn response(&self, req: &HttpRequest<S>, mut resp: HttpResponse) -> Result<Response, Error> {
+        resp.headers_mut().insert(
+            header::HeaderName::try_from("Content-Security-Policy").unwrap(),
```

Comment:
What about actix_web::middleware::DefaultHeaders (it basically does this)?

if this middleware is necessary:

unfortunately this does require the HeaderValue::from_static I think but you don't need the extra bits for the keys:

```suggestion
            "Content-Security-Policy",
```

Reply:
thanks! looks like i can pass in str for both values.